### PR TITLE
Add support for brotli encoding

### PR DESCRIPTION
**Pull request by thornjad**
As requested in #445, this adds support for serving brotli-encoded files by passing `--brotli` or `-b` to http-server. This necessitates bumping up the minimum `ecstatic` dependency version to `3.3.0`, but this [shouldn't have any breaking changes](https://github.com/jfhbrook/node-ecstatic/blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by filipesilva** [comment]
@indexzero @BigBlueHat anything I can help out with to get this merged?

I tested it locally by following these steps:
- clone this repo
- copy `package-lock.json` to `public/package-lock.json`
- encode `public/package-lock.json` (78kb) to `public/package-lock.json.br` (17kb) via https://dominikhlbg.github.io/brotlijs/
- run `./bin/http-server`
- open chrome at http://localhost:8080/package-lock.json, open devtools-network, ctrl+f5, see original package-lock size
- terminate previous process, run `./bin/http-server -b`
- open chrome at http://localhost:8080/package-lock.json, open devtools-network, refresh, see reduced package-lock size

I think this PR is working as intended 👍

**Comment by filipesilva** [comment]
Heya, don't mean to be a bother but just wanted to ping about this again since there's a project I'm working on that's waiting for this support. Keep up the good work!

**Comment by thornjad** [comment]
@BigBlueHat do you have the ability to merge? And if so, what do you think?

**Comment by BigBlueHat** [comment]
@thornjad sorry I hadn't responded earlier other than the labeling.

Since this bumps the underlying `node-ecstatic` to v3.3.0, I want to be sure that doesn't have any unexpected consequences. IIRC there was some significant shift in the MIME libraries in use which may effect custom type loading for some folks.

We've also got a bit of a tangle with #464, but since you're the author of both, maybe you can help queue those in your preferred order? 😃 

Regardless, I do want to get this merged--and ideally soon.

We should also get some tests written for the `.br` and the priority order thing around `.br` priority vs. `.gzip`.

**Comment by thornjad** [comment]
Hey, that's alright. 

Tests are a good idea. I did put some in `ecstatic`, so we are partially covered that way, but `http-server` should be properly testing itself. I actually keep a sticky note on my desk that just reads "You need more tests".

For the MIME libraries, tests would probably be good here as well, but I've never used custom types in real life, so I'm not sure what to look for.

As for #464, either PR can go first and the other can be updated to match. This PR is a more meaningful change, so maybe this one should go first.

**Comment by BigBlueHat** [comment]
@thornjad wrt to mime stuff, I may be remembering this all wrong.  It looks like #35 is a pending request which has an old PR #152 which may be obviated by `node-ecstatic`'s direct [mime type customization](https://github.com/jfhbrook/node-ecstatic#optsmimetypes).

PR #472 exposes the raw JSON mime config to the command line, but doesn't handle the `.types` loading which `node-ecstatic` makes available as a path set on `--mime-types`.

In short, this is probably not a blocker. 😄 But it is something there's long standing desire for, so maybe we tackle that once much of this is in place.

With some basic tests for this code, I'd be say it's ready to go (it reads well anyway).

Thanks for the help @thornjad!

**Comment by wentjun** [comment]
Hi! I have been following this issue for a while. May I know when will this be merged/released?

**Comment by thornjad** [comment]
@wentjun sorry for the delay! I am working on a giant merge train so this should be merged soon!

**Comment by filipesilva** [comment]
Awesome, happy to see this go in!
